packages/client/src/components/Component.svelte

```diff
@@ -650,6 +664,7 @@
     data-name={name}
     data-icon={icon}
     data-parent={$component.id}
+    use:gridLayout={gridMetadata}
```

Comment:
Most of the new metadata required to use this feature is handled by this new svelte action. This keeps it all nicely scoped inside one file and means we don't need to make this core Component file any more complicated than it already is.